src/main/groovy/org/burnmydread/SimpleRandom.groovy

```diff
@@ -12,12 +12,36 @@ import java.nio.ByteBuffer
  * The intent is to have a easy to use Secure RNG.
  */
 class SimpleRandom {
+    /* MP COMMENT
+     * Couple of points here:
+     * 1. This throws an IOException that needs to be handled, so at a minimum you'll need
+     *    a constructor for this class, although IO should generally not be done in a constructor
+     *    either.  Even better would be have an init method that initializes its state, but
+     *    that makes the API a little harder to use.
```

Comment:
"although IO should generally not be done in a constructor". I am not sure where this is coming from. I agree with the rest of what you are saying, but I would like to know why you think this. My general stance is that you should fail as quickly as possible, or not at all. At the moment I am only targeting unix platforms, so if we don't have /dev/random or /dev/urandom we should stop in the constructor. The long term goal is to support windows, so I can see adding a fail safe to fall back on SecureRandom but I want to get the methodology around figuring out if we have ok entropy first.